### PR TITLE
chore: Convert `desktop/lib` to TypeScript

### PR DESCRIPTION
**Pull request by icirellik**
Converts 3 files in `desktop/lib` to TypeScript and removes one unused
file.

### CODE REVIEW

File: src/desktop/apps/consign/index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import * as routes from "./routes"
 import JSONPage from "../../components/json_page"
-import adminOnly from "../../lib/admin_only"
+// import adminOnly from "../../lib/admin_only"
```
Comment (eessex) [inline src/desktop/apps/consign/index.js:3]: unintentional comment?

File: src/desktop/test/lib/admin_only.test.js [hunk 2]
```diff
@@ -1,35 +1,31 @@
...
+    beforeEach(() => {
+      req = { user: new Backbone.Model({ type: "Admin" }) }
     })
 
-    return it("passes through without error", function () {
-      adminOnly(this.req, {}, this.next)
-      return _.isUndefined(this.next.args[0][0]).should.be.true()
+    it("passes through without error", () => {
+      adminOnly(req, {}, next)
+      _.isUndefined(next.args[0][0]).should.be.true()
```
Comment (eessex) [inline src/desktop/test/lib/admin_only.test.js:21]: We should update this assertion to remove underscore dependency.

File: src/desktop/test/lib/admin_only.test.js [hunk 3]
```diff
@@ -1,35 +1,31 @@
...
 
-describe("adminOnly middleware", function () {
-  beforeEach(function () {
-    return (this.next = sinon.stub())
+describe("adminOnly middleware", () => {
+  let next
+  let req
+
+  beforeEach(() => {
+    next = sinon.stub()
```
Comment (eessex) [inline src/desktop/test/lib/admin_only.test.js:11]: This looks like it would be a pretty straightforward test to convert to jest and remove sinon.

Comment (icirellik) [reply src/desktop/test/lib/admin_only.test.js:11]: Definitely, test conversions weren't the original intent for this PR, I can quickly convert the tests in the `src/desktop/test/lib/` folder if you'd like.

**Pull request merged**